sql/database.go

```diff
@@ -51,3 +53,39 @@ func (p *planner) getDatabaseDesc(name string) (*DatabaseDescriptor, error) {
 	}
 	return &desc, nil
 }
+
+// getCachedDatabaseDesc looks up the database descriptor given its name in the
+// descriptor cache.
+func (p *planner) getCachedDatabaseDesc(name string) (*DatabaseDescriptor, error) {
+	if name == SystemDB.Name {
+		return &SystemDB, nil
+	}
+
+	nameKey := databaseKey{name}
+	var nameVal *roachpb.Value
+	if p.systemConfig != nil {
```

Comment:
And yet we never call into this code path for such planners. I've removed the nil check.